napari/components/add_layers_mixin.py

```diff
@@ -44,6 +50,65 @@ def add_layer(self, layer):
         if len(self.layers) == 1:
             self.reset_view()
 
+    def add_path(
+        self, path: Union[str, Sequence[str]], stack: bool = False
+    ) -> List[layers.Layer]:
+        """Add a path or list of paths to the viewer.
+
+        A list of paths will be handed one-by-one to the napari_get_reader hook
+        if stack is False, otherwise the full list is passed to each plugin
+        hook.
+
+        Parameters
+        ----------
+        path : str or list of str
+            A filepath, directory, or URL (or a list of any) to open.
+        stack : bool, optional
+            If a list of strings is passed and ``stack`` is ``True``, then the
+            entire list will be passed to plugins.  It is then up to individual
+            plugins to know how to handle a list of paths.  If ``stack`` is
+            ``False``, then the ``path`` list is broken up and passed to plugin
+            readers one by one.  by default False.
+
+        Returns
+        -------
+        layers : list
+            A list of any layers that were added to the viewer.
+        """
+        paths = [path] if isinstance(path, str) else path
+        paths = [fspath(path) for path in paths]  # PathObjects -> str
+        if not isinstance(paths, (tuple, list)):
+            raise ValueError(
+                "'path' argument must be a string, list, or tuple"
+            )
+
+        added: List[layers.Layer] = []  # for layers that get added
+
+        def load(_path):
```

Comment:
I find this nested function a bit confusing/hard to follow. Some suggested changes:

- rename `_path` to `_path_or_paths`
- add comment to the section `if not layer_data`: when this happens (plugin can't read), and what the response is (nothing much, but it looks like a lot of code)

As a bonus, I don't see it clearly, but it feels to me like the nested function definition isn't necessary. It might be feasible to make the code clearer e.g. by having an *external* rather than nested function.

Reply:
Will do. I considered making it a separate function, and I'm fine with that.  It will probably make it a private module-level function (or throw it in utils), since it doesn't need `self` at all... unless you object to that.